app/screens/browse_channels/browse_channels.tsx

```diff
@@ -0,0 +1,501 @@
+    }, []);
+
+    const renderItem = useCallback(({item}: {item: Channel}) => {
+        return (
+            <ChannelListRow
+                channel={item}
+                testID='more_channels.custom_list.channel_item'
+                onPress={onSelectChannel}
+            />
+        );
+
+    // All dependencies derived from onSelectChannel
+    }, [theme, canCreateChannels, currentUserId, currentTeamId]);
+
+    const renderLoading = useCallback(() => {
+        return (
+            <Loading
+                containerStyle={style.loadingContainer}
+                style={style.loading}
+                color={theme.buttonBg}
+            />
+        );
+
+    //Style is covered by the team
+    }, [theme]);
+
+    const renderNoResults = useCallback(() => {
+        if (term) {
+            return (
+                <View style={style.noResultContainer}>
+                    <FormattedText
+                        id='mobile.custom_list.no_results'
+                        defaultMessage='No Results'
+                        style={style.noResultText}
+                    />
+                </View>
+            );
+        }
+
+        return (
+            <View style={style.noResultContainer}>
+                <FormattedText
+                    id='more_channels.noMore'
+                    defaultMessage='No more channels to join'
+                    style={style.noResultText}
+                />
+            </View>
+        );
+    }, [style, term]);
+
+    const renderSeparator = useCallback(() => (
+        <View
+            style={{
+                height: 1,
+                backgroundColor: changeOpacity(theme.centerChannelColor, 0.08),
+                width: '100%',
+            }}
+        />
+    ), [theme]);
+
+    useEffect(() => {
+        mounted.current = true;
+        doGetChannels();
+        return () => {
+            mounted.current = false;
+        };
+    }, []);
+
+    useEffect(() => {
+        const unsubscribe = Navigation.events().registerComponentListener({
+            navigationButtonPressed: ({buttonId}: { buttonId: string }) => {
+                switch (buttonId) {
+                    case 'close-more-channels':
+                        close();
+                        break;
+                    case 'create-pub-channel': {
+                        // TODO part of https://mattermost.atlassian.net/browse/MM-39733
+                        // Update this to use the proper constant and the proper props.
+                        const screen = 'CreateChannel';
+                        const title = intl.formatMessage({id: 'mobile.create_channel.public', defaultMessage: 'New Public Channel'});
+                        const passProps = {
+                            channelType: General.OPEN_CHANNEL,
+                            categoryId,
+                        };
+
+                        // Go to or show modal?
+                        goToScreen(screen, title, passProps);
```

Comment:
As the comment says, this originally was using "goToScreen". Not sure whether it should show a modal, go to another screen or other navigation function.

Reply:
yes, this should remain as goToScreen, basically this screen is shown as a modal (and a new stack) so every screen pushed on top of this modal is part of the stack